runtime/src/protocol.rs

```diff
@@ -62,6 +64,8 @@ pub struct Protocol {
     last_request_id: AtomicUsize,
     /// Pending outgoing requests.
     pending_out_requests: Mutex<HashMap<u64, channel::Sender<Body>>>,
+    /// Runtime ID
+    pub runtime_id: Mutex<Option<RuntimeId>>,
```

Comment:
Don't make this public, instead add an accessor that returns a `RuntimeId` (or panics if one is not available -- with corresponding note of this in the documentation block).